### PR TITLE
F/any of

### PR DESCRIPTION
**Pull request by karlkeskyla**
Functionality to support `anyOf`.

https://swagger.io/docs/specification/data-models/oneof-anyof-allof-not/

### CODE REVIEW

**Comment by karlkeskyla** [comment]
Pipeline passes for Node 12, but fails for Node 10 due to [Promise.allSettled()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled).

**Review by ssaarela — APPROVED** [review]
Looks good!

Loosely related to this: We should device a better way of reporting anyOf and oneOf errors. Violations and maybe extra successes for oneOf should be somehow grouped by (nested)validator. Without such grouping the error doesn't really help the user in fixing those errors 🤔 I'll open a feature request for this.

File: packages/core/src/validators.ts [hunk 1]
```diff
@@ -158,6 +158,12 @@ export class PatternViolation extends Violation {
   }
 }
 
+export class AnyOfMismatch extends Violation {
```
Comment (ssaarela) [inline packages/core/src/validators.ts:161]: AnyOfMismatch is not actually used anywhere, is it?

**Review by ssaarela — CHANGES_REQUESTED** [review]
Missed that one "bug": AnyOfMismatch is only referred to in documentation... which is actually misleading.

File: packages/core/src/validators.ts (hunk 1 above)
Comment (karlkeskyla) [reply packages/core/src/validators.ts:161]: Yeah, that is true. Will remove it at once

**Review by ssaarela — APPROVED** [review]

**Pull request merged**